### DISCUSSION TITLE
MooseMesh

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#18024] [Q&A General] [answered]
When running my model I get the following error:

" *** ERROR ***
/home/twise/projects/moose/modules/combined/PWR_Studies/peacock_run_exe_tmp_fsi_fallingBal.i:174: (InterfaceKernels/penalty_interface_x/boundary):
    the following side set ids do not exist on the mesh: 5

    MOOSE distinguishes between "node sets" and "side sets" depending on whether
    you are using "Nodal" or "Integrated" BCs respectively. Node sets corresponding
    to your side sets are constructed for you by default.

    Try setting "Mesh/construct_side_list_from_node_list=true" if you see this error.
    Note: If you are running with adaptivity you should prefer using side sets. "

For MooseMesh, what file(s) do I need to access and modify in order to incorporate either of these changes ?:
Mesh/construct_node_list_from_side_list=true
Mesh/construct_side_list_from_node_list=false

**Comment by GiudGiud** [comment, marked as answer]
It's likely not the issue.
What is boundary set to in penalty_interface_x ?
Could you please open your mesh in paraview and check that the boundary you are using is really there?

**Reply by TLWise** [reply]
The boundary for the penalty_interface_x is master0_interface. This boundary shows up in peacock. I believe the issue may be related to the fact that I have two separate mesh domains that do not share any side sets. The near neighbor nodes shared with each domain would probably have to be used to assemble a fictitious coupled side set.

**Reply by GiudGiud** [reply]
Can you double check the spelling of the interface name? That's a common reason too.
Then could you please add `show_info = true` to your last mesh generator (a FileMeshGenerator if you are just loading a mesh) and report the result.
InterfaceKernels are pretty good at disjoint boundaries. I m not sure it can be empty in between the boundaries, we ll ask someone after that.

**Reply by TLWise** [reply]
Here is the mesh generator information:

Mesh Information:
    elem_dimensions()={2}
    spatial_dimension()=2
    n_nodes()=21069
      n_local_nodes()=21069
    n_elem()=20608
      n_local_elem()=20608
      n_active_elem()=20608
    n_subdomains()=2
    n_partitions()=1
    n_processors()=1
    n_threads()=1
    processor_id()=0
    is_prepared()=true
    is_replicated()=true
  
   Mesh Bounding Box:
    Minimum: (x,y,z)=(       0,        0,        0)
    Maximum: (x,y,z)=(    0.31,   2.5625,        0)
    Delta:   (x,y,z)=(    0.31,   2.5625,        0)
  
   Mesh Element Type(s):
    QUADSHELL4
  
   Mesh Nodesets:
    None
  
   Mesh Sidesets:
    Sideset 1 (Walls), 788 sides (EDGE2), 788 elems (QUADSHELL4), 790 nodes
     Side volume: 5.125
     Bounding box minimum: (x,y,z)=(       0,        0,        0)
     Bounding box maximum: (x,y,z)=(    0.31,   2.5625,        0)
     Bounding box delta: (x,y,z)=(    0.31,   2.5625,        0)
    Sideset 2 (Inlet), 52 sides (EDGE2), 52 elems (QUADSHELL4), 53 nodes
     Side volume: 0.31
     Bounding box minimum: (x,y,z)=(       0,        0,        0)
     Bounding box maximum: (x,y,z)=(    0.31,        0,        0)
     Bounding box delta: (x,y,z)=(    0.31,        0,        0)
    Sideset 3 (Outlet), 52 sides (EDGE2), 52 elems (QUADSHELL4), 53 nodes
     Side volume: 0.31
     Bounding box minimum: (x,y,z)=(       0,   2.5625,        0)
     Bounding box maximum: (x,y,z)=(    0.31,   2.5625,        0)
     Bounding box delta: (x,y,z)=(    0.31,        0,        0)
    Sideset 4 (BallWall), 26 sides (EDGE2), 26 elems (QUADSHELL4), 26 nodes
     Side volume: 0.194305
     Bounding box minimum: (x,y,z)=(   0.119,  1.51923,        0)
     Bounding box maximum: (x,y,z)=(   0.181,  1.58077,        0)
     Bounding box delta: (x,y,z)=(   0.062, 0.061548,        0)
  
   Mesh Edgesets:
    None
  
   Mesh Subdomains:
    Subdomain 1 (Fluid): 20488 elems (QUADSHELL4, 20488 active), 20935 active nodes
     Volume: 0.794375
     Bounding box minimum: (x,y,z)=(       0,        0,        0)
     Bounding box maximum: (x,y,z)=(    0.31,   2.5625,        0)
     Bounding box delta: (x,y,z)=(    0.31,   2.5625,        0)
    Subdomain 2 (Particle): 120 elems (QUADSHELL4, 120 active), 134 active nodes
     Volume: 0.00298977
     Bounding box minimum: (x,y,z)=(   0.119,  1.51923,        0)
     Bounding box maximum: (x,y,z)=(   0.181,  1.58077,        0)
     Bounding box delta: (x,y,z)=(   0.062, 0.061548,        0)
    Global mesh volume = 0.797365

**Reply by GiudGiud** [reply]
So as you can see in the sidesets, moose does not know about master_0interface

You could figure out what happened to it, or if possible you could re-create it using MeshGenerators

**Reply by TLWise** [reply]
> Here is the mesh definition that I am currently using.: 
> 
> "[Mesh]
>   [basic_mesh]
>     type = FileMeshGenerator
> "#    FileMeshGenerator RenameBlockGenerator
>     file = FallingBall.e
>     show_info = true
>   []
> "#    uniform_refine = 3
> 
>" #  [subdomain1]
> "#    type = SubdomainBoundingBoxGenerator
> "#    bottom_left = '0.119 1.51923 0'
> "#    block_id = Particle_2
> "#    top_right = '0.181 1.58077 0'
> "#    input = basic_mesh
> "#  []
> 
>   [interface]
>     type = SideSetsBetweenSubdomainsGenerator
>     primary_block = Fluid
>     paired_block = Particle
>     new_boundary = 'coupled_bound'
> "#    input = subdomain
>     input = basic_mesh
>     show_info = true
>   []
> 
>   [break_boundary]
>     type = BreakBoundaryOnSubdomainGenerator
>     input = interface
>   []
> []"

Using the code generates this error:

*** ERROR ***
/home/twise/projects/moose/modules/combined/PWR_Studies/peacock_run_exe_tmp_fsi_fallingBal.i:176: (InterfaceKernels/penalty_interface_x/boundary):
    the following side set ids do not exist on the mesh: 9

    MOOSE distinguishes between "node sets" and "side sets" depending on whether
    you are using "Nodal" or "Integrated" BCs respectively. Node sets corresponding
    to your side sets are constructed for you by default.

    Try setting "Mesh/construct_side_list_from_node_list=true" if you see this error.
    Note: If you are running with adaptivity you should prefer using side sets.

**Reply by GiudGiud** [reply]
Please format the input using markdown (with quotes)

What sideset are you using for penalty_interface_x/boundary?
Can you paste the show_info output again, for the last mesh generator you used?

Thanks

**Reply by TLWise** [reply]
" [InterfaceKernels]
  [./penalty_interface_x]
    type = CoupledPenaltyInterfaceDiffusion
    variable = vel_x
    neighbor_var = disp_x
    secondary_coupled_var = vel_x_solid
"#    boundary = 'BallWall'    
    boundary = 'coupled_bound'
    penalty = 1e6
  [../]
  [./penalty_interface_y]
    type = CoupledPenaltyInterfaceDiffusion
    variable = vel_y
    neighbor_var = disp_y
    secondary_coupled_var = vel_y_solid
"#    boundary = 'BallWall'
    boundary = 'coupled_bound'
    penalty = 1e6
  [../]
[] "

**Reply by TLWise** [reply]
> Please format the input using markdown (with quotes)

Would you please provide a visual example for properly formatting to reduce the font size?

Thank you.

**Reply by TLWise** [reply]
Please find attached my input files.

[FallingBall.e.txt](https://github.com/idaholab/moose/files/6620229/FallingBall.e.txt)
[fsi_fallingBal.i.txt](https://github.com/idaholab/moose/files/6620230/fsi_fallingBal.i.txt)

**Reply by GiudGiud** [reply]
```
type = CoolObject
```
so \`\`\` before and after the text in markdown

**Reply by GiudGiud** [reply]
you can edit my comment to have a look

**Reply by GiudGiud** [reply]
show_info on [interface] should show whether coupled_bound is being created or not

**Reply by TLWise** [reply]
The show_info used on [interface] does not show the creation of the coupled_bound.

[interface]
type = SideSetsBetweenSubdomainsGenerator
primary_block = Fluid
paired_block = Particle
new_boundary = 'coupled_bound'
"#    input = subdomain"
input = basic_mesh
show_info = true 
 []

**Reply by GiudGiud** [reply]
Ok I checked the mesh, the problem is that the two domain meshes overlap. 
The fluid mesh needs to go around the solid, then it ll be modified every time step to adjust to the ball movement. 

There is no sideset between fluid and particle in the current configuration, since they do not share any element side / node

**Comment by TLWise** [comment]
Will the movement of the ball deform the mesh of the fluid, introducing numerical error?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by GiudGiud** [reply]
@fdkong 
I think the error would go down with small timesteps

**Comment by TLWise** [comment]
If the mesh domains were to be separated by some distance with no overlap would the method work in that case?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by GiudGiud** [reply]
Not an expert in FSI but doubt that would help. Would make it harder to match the sides
